### PR TITLE
Account model and controller

### PR DESCRIPTION
**Pull request by hpabst**
The User model and UserController are finished, tested, and ready for review.

### CODE REVIEW

File: TextCrunch/UserController.swift [hunk 1]
```diff
@@ -12,55 +12,117 @@ import Foundation
 //The base UserController class for non-social network user management.
 class UserController{
     
-    let LOGINFAIL_MISC = -1 //Login error code for unknown error.
-    let LOGINFAIL_PASSWORD = 0 //Login error code for if the login password is not correct.
-    let LOGINFAIL_NEXIST = 1 //Login error code for if the account does not exist.
-    let LOGINFAIL_EMAIL = 2 //Login error code for if the email is incorrect.
-    let LOGINSUCCESS = 3 //Login code for a successful login.
+    enum UCCode : Int{
```
Comment (derekdowling) [inline TextCrunch/UserController.swift]: Nice! Good to know you can do this.

File: TextCrunch/UserController.swift [hunk 2]
```diff
@@ -12,55 +12,117 @@ import Foundation
...
+        return resultCode
     }
     
-    func getCurrentUser()-> UserModel?{
-        return nil
+    
+    //Returns a User representing the currently logged-in user, or nil if no user is logged in.
+    func getCurrentUser()-> User?{
+        var currentPFUser = PFUser.currentUser()
+        if (currentPFUser != nil){
```
Comment (derekdowling) [inline TextCrunch/UserController.swift]: I honestly don't know if we really need to decouple this. I think it's going to turn into more work for us than it needs to be. All so we can make it easier for some developer decouple this later on?

File: TextCrunch/UserController.swift [hunk 3]
```diff
@@ -12,55 +12,117 @@ import Foundation
...
+    func logoutCurrentUser() -> Bool{
+        PFUser.logOut()
+        var currentUser = PFUser.currentUser()
+        return (currentUser == nil)
     }
     
-    func resetPassword(user: UserModel) -> Bool{
-        return false
+    //Initiates password reset with Parse by sending the user a password reset email.
+    func resetCurrentUserPassword() -> Void{
```
Comment (derekdowling) [inline TextCrunch/UserController.swift]: Nice, I know we didn't really account for this in the User Stories but it was something we really need to provide.

File: TextCrunch/UserController.swift (hunk 2 above)
Comment (hpabst) [inline TextCrunch/UserController.swift]: The work is already done though, and we will be handing this off to some other development team in the future.

Comment (derekdowling) [inline TextCrunch/UserController.swift]: Okay, well I guess we can change it later on if it feels really clunky.

**Comment by derekdowling** [comment]
This looks fine. Lets get it in so we can start making use of that object. @hpabst fix up your merge conflicts and feel free to merge.

**Pull request merged**